cardano-node/src/Cardano/Common/Parsers.hs

```diff
@@ -56,20 +63,46 @@ parseProtocol :: Parser Protocol
 parseProtocol = asum
   [ fl ByronLegacy "byron-legacy"
     "Byron/Ouroboros Classic suite of algorithms"
-
   , fl BFT "bft"
     "BFT consensus"
-
   , fl Praos "praos"
     "Praos consensus"
-
   , fl MockPBFT "mock-pbft"
     "Permissive BFT consensus with a mock ledger"
-
   , fl RealPBFT "real-pbft"
     "Permissive BFT consensus with a real ledger"
   ]
 
+parseProtocolByron :: Parser (Last Protocol)
+parseProtocolByron =
+  flag' (Last $ Just ByronLegacy) $ mconcat [ long "byron-legacy"
+                                            , help "Byron/Ouroboros Classic suite of algorithms"
+                                            ]
+
+parseProtocolBFT :: Parser (Last Protocol)
+parseProtocolBFT =
+  flag' (Last $ Just BFT) $ mconcat [ long "bft"
+                                    , help "BFT consensus"
+                                    ]
+
+parseProtocolPraos :: Parser (Last Protocol)
+parseProtocolPraos =
+  flag' (Last $ Just Praos) $ mconcat [ long "praos"
+                                      , help "Praos consensus"
+                                      ]
+
+parseProtocolMockPBFT :: Parser (Last Protocol)
+parseProtocolMockPBFT =
+  flag' (Last $ Just MockPBFT) $ mconcat [ long "mock-pbft"
+                                         , help "Permissive BFT consensus with a mock ledger"
+                                         ]
+
+parseProtocolRealPBFT :: Parser (Last Protocol)
+parseProtocolRealPBFT =
+  flag' (Last $ Just RealPBFT) $ mconcat [ long "real-pbft"
+                                         , help "Permissive BFT consensus with a real ledger"
+                                         ]
+
```

Comment:
Now, this is a very sparse encoding of something that is really very simple and compact -- and this sparseness is really hiding the simplicity.

Do you think we can optimise the presentation here?

Reply:
@deepfire Yep this can be condensed. I'll use `fl`.

Reply:
@deepfire pushed changes